src/custom/utils/price.ts

```diff
@@ -86,7 +86,23 @@ export type AllPricesResult = {
  */
 export async function getAllPrices(params: LegacyPriceQuoteParams): Promise<AllPricesResult> {
   // Get price from all API: Gpv2, Paraswap, Matcha (0x)
-  const gpPricePromise = withTimeout(getPriceQuoteGp(params), PRICE_API_TIMEOUT_MS, 'GPv2: Get Price API')
+
+  const cowSwapQuote = getFullQuote({
+    quoteParams: {
+      ...params,
+      isEthFlow: params.baseToken.toLowerCase() === NATIVE_CURRENCY_BUY_ADDRESS.toLowerCase(),
+      sellToken: params.baseToken,
+      buyToken: params.quoteToken,
+    },
+  }).then(([result]) => {
+    if (result.status === 'fulfilled') {
+      return result.value
+    }
+
+    return Promise.reject(result.reason)
+  })
+
+  const gpPricePromise = withTimeout(cowSwapQuote, PRICE_API_TIMEOUT_MS, 'GPv2: Get Price API')
```

Comment:
nit: I think we dont't need the GPv2 name any more. Can be named CoW Protocol

Reply:
Fixed